src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -265,16 +270,27 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_projection_node(
   // TODO(anybody) For columns newly created by a Projection no meaningful statistics can be generated yet, hence an
   //               empty AttributeStatistics object is created.
 
-  auto column_statistics =
-      std::vector<std::shared_ptr<BaseAttributeStatistics>>{projection_node.output_expressions().size()};
+  const auto& output_expressions = projection_node.output_expressions();
+  const auto output_expression_count = output_expressions.size();
+  auto column_statistics = std::vector<std::shared_ptr<BaseAttributeStatistics>>{output_expression_count};
+
+  auto matched_expressions = ExpressionUnorderedSet{};
+  projection_node.left_input()->iterate_output_expressions(
+      [&](const auto input_column_id, const auto& input_expression) {
+        for (size_t expression_idx{0}; expression_idx < output_expression_count; ++expression_idx) {
+          const auto& expression = output_expressions[expression_idx];
+          if (*expression == *input_expression) {
+            column_statistics[expression_idx] = input_table_statistics->column_statistics[input_column_id];
+            matched_expressions.emplace(expression);
```

Comment:
Early `break/return`?

Reply:
Same problem as early return for alias expressions, but for projections. We get a segfault for the same projection of the same SQLite test line, as it contains columns that originate from the same stored table node column.